common/src/main/java/org/apache/iceberg/common/DynConstructors.java

```diff
@@ -67,7 +66,6 @@ public C newInstance(Object... args) {
       try {
         return newInstanceChecked(args);
       } catch (Exception e) {
-        Throwables.propagateIfInstanceOf(e, RuntimeException.class);
```

Comment:
I do not think it is safe to remove.
The result is that the `RuntimeException`s will be wrapped in a new exception which will cause issues with error handling